org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java

```diff
@@ -318,8 +318,8 @@ private Optional<Symbol> symbol(JCTree value) {
 		if (value instanceof JCTree.JCMethodDecl jcMethodDecl) {
 			return Optional.ofNullable(jcMethodDecl.sym);
 		}
-		if (value instanceof JCTree.JCTypeParameter jcTypeParam) {
-			return Optional.ofNullable(jcTypeParam.type.tsym);
+		if (value instanceof JCTree.JCTypeParameter jcTypeParam && jcTypeParam.type.tsym instanceof Symbol.TypeVariableSymbol typeVariableSymbol) {
```

Comment:
I don't get how this would prevent an NPE. Is the NPE happening when dereferencing null `type` to get `.tsym` field? If so, the NPE would still happen, just 1 line before.

Reply:
You're right. I don't think this change will fix the NPE. I'll spend some more time looking into it.